packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -482,27 +503,32 @@ export class OdspDocumentStorageService extends OdspDocumentStorageServiceBase {
 
 	private async fetchSnapshot(
 		hostSnapshotOptions: ISnapshotOptions | undefined,
+		groupIds: string[],
 		scenarioName?: string,
 	) {
-		return this.fetchSnapshotCore(hostSnapshotOptions, scenarioName).catch((error) => {
-			// Issue #5895:
-			// If we are offline, this error is retryable. But that means that RetriableDocumentStorageService
-			// will run in circles calling getSnapshotTree, which would result in OdspDocumentStorageService class
-			// going getVersions / individual blob download path. This path is very slow, and will not work with
-			// delay-loaded data stores and ODSP storage deleting old snapshots and blobs.
-			if (typeof error === "object" && error !== null) {
-				error.canRetry = false;
-			}
-			throw error;
-		});
+		return this.fetchSnapshotCore(hostSnapshotOptions, groupIds, scenarioName).catch(
+			(error) => {
+				// Issue #5895:
+				// If we are offline, this error is retryable. But that means that RetriableDocumentStorageService
+				// will run in circles calling getSnapshotTree, which would result in OdspDocumentStorageService class
+				// going getVersions / individual blob download path. This path is very slow, and will not work with
+				// delay-loaded data stores and ODSP storage deleting old snapshots and blobs.
+				if (typeof error === "object" && error !== null) {
+					error.canRetry = false;
+				}
+				throw error;
+			},
+		);
 	}
 
 	private async fetchSnapshotCore(
 		hostSnapshotOptions: ISnapshotOptions | undefined,
+		groupIds: string[],
 		scenarioName?: string,
 	): Promise<ISnapshot | IPrefetchSnapshotContents> {
-		// Don't look into cache, if the host specifically tells us so.
-		if (!this.hostPolicy.avoidPrefetchSnapshotCache) {
+		// Don't look into cache, if the host specifically tells us so. Also, if request is
+		// for initial snapshot, don't consult the prefetch cache.
+		if (!this.hostPolicy.avoidPrefetchSnapshotCache && this.firstVersionCall) {
```

Comment:
is dropping this.firstVersionCall here intentional? I feel like nothing should have changed here with new flow, i.e. only first call should consult cache.

Reply:
Previously, only the first getVersions call used to have this flow, so it was not required. Not other calls will also come here(for specific groupids), so we need to include this in "if" now.